### PR TITLE
Fix hookFilterProductContent to allow hook chain

### PR DESCRIPTION
**Pull request by psandmore**
All hookFilter* functions should always return the following:
    return $params;
Returning $params ensures that other modules in the same hook chain receive the correct $params argument.
To reproduce the bug install an additional module with hookFilterProductContent AFTER the productcomments and run in debug mode.
The following exception appears.
Type error: Argument 3 passed to HookCore::callHookOn() must be of the type array, null given, called in .../classes/Hook.php on line 934

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | All hookFilter* functions should always return the following:  return $params; Returning $params ensures that other modules in the same hook chain receive the correct $params argument.
| Type?         | bug fix / critical
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | To reproduce the bug install an additional module with hookFilterProductContent AFTER the productcomments and run in debug mode.

### CODE REVIEW

**Comment by idnovate** [comment]
Please merge this PR 👍

**Comment by PierreRambaud** [comment]
> Please merge this PR +1

We will not merge it until the CI is fixed :thinking:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Could you please properly fill the description table and fix the CI?

**Comment by psandmore** [comment]
Fixed CI

**Comment by kpodemski** [comment]
@psandmore You didn't send any changes

**Comment by psandmore** [comment]
There are changes in one file that I committed and a clear explanation what has been changed. PrestaShop is YOUR product. If you like YOUR product to be better, please be more flexible and merge the changes. I am tired with this ping-pong. I do not need this critical bug fix - you need it. If you also do not need people contributing to your product, don't ask others to help you.

**Comment by kpodemski** [comment]
@psandmore 

There's no need to be angry. We really appreciate your contribution. The thing is, you told us you fixed CI, but you didn't make any changes since the initial commit. Maybe you forgot to do `git push`? We cannot accept a contribution that doesn't meet the requirements and standards of the project.

> PrestaShop is YOUR product. If you like YOUR product to be better, please be more flexible and merge the changes. 

It's an open-source project used by hundreds of thousands of people. Sometimes we can't be flexible. We can't be flexible when our tools check for quality code and report that something is wrong.

**Comment by JBWModules** [comment]
Please merge this PR, without it the productcomment module breaks one of my modules (and probably all others) that uses the same hook!
I don't know what is CI and how to fix it, but if you explain I will be happy to support.

**Comment by idnovate** [comment]
https://github.com/PrestaShop/productcomments/pull/137

**Comment by psandmore** [comment]
I committed changes to the code. What else changes do you need, @kpodemski?

**Comment by psandmore** [comment]
Looks like PrestaShop do not need work from contributors. Shame on you, PrestaShop!

**Comment by matks** [comment]
@idnovate did the right thing: he submitted PR https://github.com/PrestaShop/productcomments/pull/137 which passes CI

> I don't know what is CI and how to fix it, but if you explain I will be happy to support.

Here @idnovate did a great job 👍 but next time you want to explore CI, let's do it together on the Slack channel 😉 it's great to see you're motivated to learn

@psandmore You do not understand.

> I do not need this critical bug fix - you need it.

I personnally don't need it. If I need something, I do it 😄 . This is how open source works. [Your Pull Request is not a gift](https://dev.to/matks/your-pull-request-is-not-a-gift-237m).

People contribute to the topics that are important to them. And all of this is merged together and everybody benefit from each individual effort.

When I need a bug to be fixed, I fix it. I do not wait for someone else to do it. The maintainer team cannot fix everything so we choose the ones we fix (usually the critical and major bugs) and we let motivated people choose the bugs that are important to them and fix them :) so everybody pushes what's important to her/him and at the end everybody benefit from everybody's work. Read [the power of multiplication](https://dev.to/matks/why-should-i-contribute-to-open-source-21p) to see more about it.

> If you like YOUR product to be better, please be more flexible and merge the changes.

What you're pushing for is shortcuts. Shortcuts reduce quality. Do you want PrestaShop to have more bugs? Do you want PrestaShop quality to decrease? I don't think so. Quality is not free. Quality is expensive

PrestaShop is not OUR product. And it's not a product, it's a project. It's EVERYBODY project. It's yours, too. If you want this project to grow you need to join the group effort and play by the rules that have been defined to make sure the software continue on the right path, and the software quality increases. And the software quality will only increase if we follow best practices such as Continuous Integration.

**Pull request closed without merging**
